proxy/src/bind.rs

```diff
@@ -231,15 +233,15 @@ impl<B> control::discovery::Bind for BindProtocol<Arc<ctx::Proxy>, B>
 where
     B: tower_h2::Body + 'static,
 {
-    type Endpoint = SocketAddr;
+    type Endpoint = Endpoint;
     type Request = http::Request<B>;
     type Response = HttpResponse;
     type Error = <Service<B> as tower::Service>::Error;
     type Service = Service<B>;
     type BindError = ();
 
-    fn bind(&self, addr: &SocketAddr) -> Result<Self::Service, Self::BindError> {
-        Ok(self.bind.bind_service(addr, &self.protocol))
+    fn bind(&self, ep: &Endpoint) -> Result<Self::Service, Self::BindError> {
```

Comment:
TIOLI, really minor nit: changing the parameter type to `Self::Endpoint` rather than explicitly naming that type means it would only have to be changed in one place if the type changes later on.